### DISCUSSION TITLE
how can i keep the form state when using the create & create another button

### DISCUSSION THREAD

**Discussion by presidenwashil** [filamentphp/filament#14383] [Help] [answered]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

I have created a form that dynamically renders the repeater items. What I want is for the form not to reset when I click the 'Create & Create another' button. how can i do that?

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Section::make(__('Meet Information'))
                    ->columns(2)
                    ->schema([
                        Select::make('group_code')
                            ->relationship('group', 'name')
                            ->label(__('Group'))
                            ->searchable()
                            ->required()
                            ->placeholder(__('Select the group'))
                            ->live(onBlur: false)
                            ->afterStateUpdated(function (Set $set, Get $get, ?string $state) {
                                if (filled($state)) {
                                    $students = Student::whereHas('groups', function ($query) use ($state) {
                                        $query->where('group_student.group_code', $state);
                                    })->get();

                                    $repeatItems = $students->map(function ($student) {
                                        return [
                                            'student_code' => $student->code,
                                            'status' => 'Hadir',
                                        ];
                                    })->toArray();

                                    $set('attendances', $repeatItems);
                                    session(['attendances' => $repeatItems]);

                                    $meetCount = Meet::where('group_code', $state)->count();
                                    $set('meet_count', $meetCount + 1);
                                }
                            }),
                        DatePicker::make('date')
                            ->label(__('Date'))
                            ->native(false)
                            ->format('Y-m-d')
                            ->placeholder(placeholder: 'Sep 24, 2024')
                            ->default(Carbon::now()->format('Y-m-d'))
                            ->required(),
                        TextInput::make('meet_count')
                            ->label(__('Meet Count'))
                            ->type('number')
                            ->placeholder(__('Enter the meet count'))
                            ->required(),
                        Select::make('period_code')
                            ->relationship('period', 'code')
                            ->getOptionLabelFromRecordUsing(fn(Period $record): string => "{$record->code} - {$record->start} - {$record->end}")
                            ->label(__('Period'))
                            ->placeholder(__('Select the period'))
                            ->searchable()
                            ->required(),
                        TextInput::make('material')
                            ->translateLabel()
                            ->placeholder(__('Enter the material')),
                        TextInput::make('note')
                            ->translateLabel()
                            ->placeholder(__('Enter the note'))
                    ]),
                Section::make(__('Attendances'))
                    ->columns(1)
                    ->schema([
                        Repeater::make('attendances')
                            ->relationship('attendances')
                            ->schema([
                                Select::make('student_code')
                                    ->relationship('student', 'name')
                                    ->label('Student')
                                    ->searchable()
                                    ->getOptionLabelUsing(function ($value) {
                                        $student = Student::find($value);
                                        return $student ? "{$student->code} - {$student->name}" : $value;
                                    })
                                    ->required(),
                                Select::make('status')
                                    ->options([
                                        'Hadir' => 'Hadir',
                                        'Izin' => 'Izin',
                                        'Sakit' => 'Sakit',
                                        'Alpa' => 'Alpa',
                                    ])
                                    ->label('Status')
                                    ->required(),
                            ])
                            ->deletable(false)
                            ->defaultItems(0)
                            ->minItems(0)
                            ->columns(2)
                    ]),
                Section::make('Assistant Attendances')
                    ->schema([
                        Repeater::make('assistantAttendances')
                            ->relationship('assistantAttendances')
                            ->simple(
                                Select::make('assistant_code')
                                    ->relationship('assistant', 'name')
                                    ->label('Assistant')
                                    ->searchable()
                                    ->required(),
                            )
                    ]),
            ]);
    }

**Comment by leandrocfe** [comment, marked as answer]
I think you can use this in the [afterCreate](https://filamentphp.com/docs/3.x/panels/resources/creating-records#lifecycle-hooks) hook

```php
protected function afterCreate(): void
{
    $this->getCreatedNotification()?->send();
    $this->halt();
}
```

**Reply by presidenwashil** [reply]
thank u very much, its working correctly
